crates/matrix-sdk/src/send_queue.rs

```diff
@@ -746,6 +787,162 @@ impl QueueStorage {
     async fn local_echoes(&self) -> Result<Vec<QueuedEvent>, RoomSendQueueStorageError> {
         Ok(self.client()?.store().load_send_queue_events(&self.room_id).await?)
     }
+
+    /// Try to apply a single dependent event, whether it's local or remote.
+    ///
+    /// This swallows errors that would retrigger every time if we retried
+    /// applying the dependent event: invalid edit content, etc.
+    #[instrument(skip_all)]
+    async fn try_apply_single_dependent_event(
+        &self,
+        client: &Client,
+        de: DependentQueuedEvent,
+    ) -> Result<(), RoomSendQueueError> {
+        let store = client.store();
+
+        match de.kind {
+            DependentQueuedEventKind::Edit { new_content } => {
+                if let Some(event_id) = de.event_id {
```

Comment:
Wait, this is the even ID of the parent event? Or why are we saying that the event has been sent, so we need to send an edit event?

Reply:
Uh, I'll update the field comment to make it clearer: `de.event_id` is the event id of the event we're depending on, not an event id for the dependent action.